net/httpclient.go

```diff
@@ -381,6 +399,7 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 }
 
 func (t *Transport) injectSpan(req *http.Request) (*http.Request, opentracing.Span) {
+	log.Println("injectSpan")
```

Comment:
injectSpan seems to be broken - it updates request context only when there is a parent span.
I think  it should be:
```diff
diff --git a/net/httpclient.go b/net/httpclient.go
index 2a4eed1a..db944f32 100644
--- a/net/httpclient.go
+++ b/net/httpclient.go
@@ -407,10 +407,10 @@ func (t *Transport) injectSpan(req *http.Request) (*http.Request, opentracing.Sp
                string(ext.HTTPUrl):    req.URL.String(),
        }}
        if parentSpan := opentracing.SpanFromContext(req.Context()); parentSpan != nil {
-               req = req.WithContext(opentracing.ContextWithSpan(req.Context(), parentSpan))
                spanOpts = append(spanOpts, opentracing.ChildOf(parentSpan.Context()))
        }
        span := t.tracer.StartSpan(t.spanName, spanOpts...)
+       req = req.WithContext(opentracing.ContextWithSpan(req.Context(), span))
 
        _ = t.tracer.Inject(span.Context(), opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(req.Header))
```

Reply:
Oh yeah that makes sense!
Only one case is now failing